src/python/python/ad/integrators/common.py

```diff
@@ -477,6 +462,30 @@ def prepare(self,
 
         return sampler, spp
 
+    def _splat_to_block(block: mi.ImageBlock,
```

Comment:
It's not really a simple splat (aka atomic-add), right? (given the weight division).

This might be more difficult, but can the whole "differentiate a write to get a read" be factored out? (That's what initially had in mind, and it would apply specifically to the reverse mode parts).

Reply:
I think we are talking about two seperate things. 

1. I factored out the block splatting (HDR vs Spec film) logic in `_splat_to_block`. This is not about differentiation, e.g. it is used in the primal rendering method as well.
2. In `render_backward()` I factored out the backpropagation code that pushes the gradient image through the splatting + weight division. You can find this code in the local function `splatting_and_backward_gradient_image`, which is used twice in `render_backward()`.

Is this more clear, or you meant that something else needs to be factored out?